src/dmd/parse.d

```diff
@@ -1440,7 +1455,7 @@ final class Parser(AST) : Lexer
     /***********************************************
      * Parse const/immutable/shared/inout/nothrow/pure postfix
      */
-    StorageClass parsePostfix(StorageClass storageClass, AST.Expressions** pudas)
+    StorageClass parsePostfix(StorageClass storageClass, AST.Expressions** pudas, Loc* udaLoc)
```

Comment:
Is there a reason to use `*` rather than `ref`?

Reply:
Or `out` ?

Reply:
used with `null` [here](https://github.com/dlang/dmd/pull/9359/files#diff-cd066d37445cac534313c0137c2d4bbeR2592)

Reply:
It's not actually relying on it being null. Supplying a temporary for the parameter will work fine, and it will remove the null dependence.

Reply:
As far as I know, using an `out` parameter would require to declare a variable at the call site, which would not be used afterwards in a couple of cases.

Reply:
That is correct. However, on the plus side:

1. It reduces the complexity of the code by eliminating the conditional logic.
2. It eliminates use of a pointer, thereby eliminating potential memory corruption.
3. It eliminates the need to document that the argument may be null.
4. It eliminates the need for the user to check that there's a null check on every dereference on the pointer.
5. `out` self-documents that this will be initialized by the function, not the caller.
6. The compiler optimizer knows (5) too, though currently it does not make use of such information.

Hence it is a better practice to use `out`.